### DISCUSSION TITLE
Question about calculating power deficit

### DISCUSSION THREAD

**Discussion by labbatessa** [NatLabRockies/floris#981] [Q&A] [answered]
How does FLORIS calculate power deficit (e.g. theta)? To provide some context, there have been papers in the literature that have calculated reference power as: Reference Power = Pₘₑₐₛᵤᵣₑ / cos²(θ), and I'm curious how FLORIS calculates θ. Thank you!

**Comment by misi9170** [comment, marked as answer]
Hi @labbatessa , 

Yes, this "cosine loss" is the standard loss model in FLORIS when there is a yaw misalignment (when the turbine is not directly aligned with the flow). [Here is the FLORIS documentation that describes that](https://nrel.github.io/floris/operation_models_user.html#cosine-loss-model). Note that $p_P$ is the cosine exponent, which is 2 in your question, but there is some debate in the literature about what the appropriate value for this should be, and FLORIS defaults to $p_P = 1.88$. However, you can change this by modifying the `cosine_loss_exponent_yaw` key on your turbine model ([here is the line to change on the NREL 5MW model](https://github.com/NREL/floris/blob/main/floris/turbine_library/nrel_5MW.yaml#L44)).

For a brief discussion on why $p_P$ defaults to 1.88 in FLORIS, you could look at Section 2.3 of [Annoni et al. (2018)](https://wes.copernicus.org/articles/3/819/2018/).

In all of these contexts, $\gamma$ (which is the same as your $\theta$) is the "error" between the turbine's absolute yaw position ($\text{AYP}$) and the wind direction ($\text{WD}$). That is, $\gamma = \theta = \text{AYP} - \text{WD}$. Note that the sign of $\gamma$ is not important for calculating the effect on turbine power due to the symmetry of the cosine about zero (however, it will affect the direction a wake deflects). In FLORIS, $\gamma$ ($\theta$) is set using the [`yaw_angles` keyword argument](https://github.com/NREL/floris/blob/main/floris/floris_model.py#L367) to `fmodel.set()`. Here, you should think of `yaw_angles` as being the _relative_ yaw angle (so that 0 represents the turbine pointing directly into the wind, whatever the wind direction may be).

You can also see the effects of yawing in [examples/examples_control_types](https://github.com/NREL/floris/tree/main/examples/examples_control_types) as well as [examples/examples_turbine/001_check_turbine.py](https://github.com/NREL/floris/blob/main/examples/examples_turbine/001_check_turbine.py).

Cheers,
Misha

**Reply by labbatessa** [reply]
Thank you @misi9170!
